### PR TITLE
Sonarqube

### PR DESCRIPTION
**Pull request by outscale-toa**

### CODE REVIEW

File: api/client/nic.cc [hunk 1]
```diff
@@ -355,7 +358,11 @@ NicAddOptions::NicAddOptions() {
     bypass_filtering = "false";
 }
 
-int NicAddOptions::parse(int argc, char **argv) {
+static inline bool IsArg(int count, int argc, char **argv, char const *option) {
```
Comment (outscale-mgo) [inline api/client/nic.cc]: mouhahahaha

nasty rogue change :)

File: api/client/nic.cc [hunk 2]
```diff
@@ -358,19 +358,23 @@ NicAddOptions::NicAddOptions() {
     bypass_filtering = "false";
 }
 
+static inline bool IsArg(int count, int argc, char **argv, char const *option) {
+    return (count + 1 < argc && string(argv[count]) == option);
```
Comment (outscale-mgo) [inline api/client/nic.cc]: I would write this
`return (count + 1) < argc && !strcmp(argv[count]), option);`
not everyone know C++ Operator Precedence, but the parentheses around return value are clary not useful.
http://en.cppreference.com/w/cpp/language/operator_precedence

File: api/client/nic.cc [hunk 3]
```diff
@@ -358,19 +358,23 @@ NicAddOptions::NicAddOptions() {
     bypass_filtering = "false";
 }
 
+static inline bool IsArg(int count, int argc, char **argv, char const *option) {
```
Comment (outscale-mgo) [inline api/client/nic.cc]: really bad name, this function doesn't not check if a string is an arg, it check if there is an arg, and if this arg correspond to the option.
maybe CheckOption

File: api/client/request.cc [hunk 4]
```diff
@@ -132,6 +132,27 @@ int Request(const string &req,
     return Request(proto_req, res, options, response_to_stdout);
 }
 
+static void PrintMessage(MessageV0_Error error) {
```
Comment (outscale-mgo) [inline api/client/request.cc]: well, this function is clary not printing A Message... but PrintError

File: api/server/api_0.cc [hunk 5]
```diff
@@ -474,10 +474,8 @@ void Api0::AppConfig(const MessageV0_Request &req, MessageV0_Response *res) {
     app::log.Info("Application configuration");
     auto config = req.app_config();
     std::string err;
-    if (config.has_log_level()) {
-        if (!app::log.SetLogLevel(config.log_level()))
-            err = err + "log level: failed ";
-    }
+    if (config.has_log_level() && (!app::log.SetLogLevel(config.log_level())))
```
Comment (outscale-mgo) [inline api/server/api_0.cc]: ` if (config.has_log_level() && !app::log.SetLogLevel(config.log_level()))`
there is an useless level of parentheses around app::log.SetLogLevel

File: api/client/request.cc [hunk 6]
```diff
@@ -156,25 +177,7 @@ int CheckRequestResult(const proto::Messages &res) {
         } else if (!res_0.response().status().status()) {
             cerr << "error in response";
             if (res_0.response().status().has_error()) {
```
Comment (outscale-mgo) [inline api/client/request.cc]: useless brace as PrintError is only one line :)

File: api/server/graph.cc [hunk 7]
```diff
@@ -653,53 +676,21 @@ std::string Graph::NicExport(const app::Nic &nic) {
...
-            std::to_string(nic.vni) + " don't seems to exist.");
-        return;
-    }
-    struct GraphVni &vni = vni_it->second;
-
-    auto nic_it = vni.nics.find(nic.id);
-    if (nic_it == vni.nics.end()) {
-        LOG_ERROR_("NIC id: " + nic.id + " in vni: " +
-            std::to_string(nic.vni) + " don't seems to exist in branch.");
+    Graph::GraphNic *graph_nic = GetGraphNic(nic);
```
Comment (outscale-mgo) [inline api/server/graph.cc]: useless braces

File: api/server/graph.cc [hunk 8]
```diff
@@ -653,53 +676,21 @@ std::string Graph::NicExport(const app::Nic &nic) {
...
-        return;
-    }
-    *in = *out = 0;
-
-    auto vni_it = app::graph.vnis_.find(nic.vni);
-    if (vni_it == app::graph.vnis_.end()) {
-        LOG_ERROR_("NIC id: " + nic.id + " in vni: " +
-            std::to_string(nic.vni) + " don't seems to exist.");
+    Graph::GraphNic *graph_nic = GetGraphNic(nic);
+    if (graph_nic == NULL) {
```
Comment (outscale-mgo) [inline api/server/graph.cc]: useless braces

File: api/server/graph.h [hunk 9]
```diff
@@ -240,6 +240,7 @@ class Graph {
        std::map<std::string, struct GraphNic> nics;
     };
 
+    GraphNic *GetGraphNic(const app::Nic &nic);
```
Comment (outscale-jju) [inline api/server/graph.h]: rename this `find` and put this in `GraphNic` class as a static method

File: api/server/graph.cc [hunk 10]
```diff
@@ -579,6 +579,29 @@ std::string Graph::NicAdd(const app::Nic &nic) {
     return std::string(ret);
 }
 
+Graph::GraphNic *Graph::GetGraphNic(const app::Nic &nic) {
+    if (!started) {
+        LOG_ERROR_("Graph has not been started");
+        return NULL;
+    }
```
Comment (outscale-jju) [inline api/server/graph.cc]: You can remove this

**Pull request closed without merging**